src/main/java/com/example/spring_bank_project/shared/infrastructure/adapter/publisher/RabbitMqReceiver.java

```diff
@@ -0,0 +1,19 @@
+package com.example.spring_bank_project.shared.infrastructure.adapter.publisher;
+
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.CountDownLatch;
+
+@Service
+public class RabbitMqReceiver {
+    private final CountDownLatch latch = new CountDownLatch(1);
+
+    public void handle(String message) {
+        System.out.println("Received <" + message + ">");
+        latch.countDown();
```

Comment:
po pierwszym wywołaniu metody handle latch staje sie bezuzyteczny, za javadocem tej metody: If the current count equals zero then nothing happens